iib/workers/tasks/build_merge_index_image.py

```diff
@@ -317,24 +317,25 @@ def is_bundle_version_valid(bundle_path, valid_ocp_version):
     v4.7   | included  | NOT included |  included   |  included    |  included
     """
     try:
-        float_valid_ocp_version = float(valid_ocp_version.replace('v', ''))
+        float_valid_ocp_version = correct_float_precision(valid_ocp_version.replace('v', ''))
     except ValueError:
         raise IIBError(f'Invalid OCP version, "{valid_ocp_version}", specified in Index Image')
     try:
         bundle_version_label = get_image_label(bundle_path, 'com.redhat.openshift.versions')
         bundle_version = bundle_version_label.replace('v', '')
+        log.debug(f'Bundle path {bundle_path} version {bundle_version_label}, VALID version {float_valid_ocp_version}')
         if bundle_version_label.startswith('='):
-            if float(bundle_version.strip('=')) == float_valid_ocp_version:
+            if correct_float_precision(bundle_version.strip('=')) == float_valid_ocp_version:
                 return True
         elif '-' in bundle_version_label:
-            min_version, max_version = [float(version) for version in bundle_version.split('-')]
+            min_version, max_version = [correct_float_precision(version) for version in bundle_version.split('-')]
             if min_version <= float_valid_ocp_version <= max_version:
                 return True
         elif "," in bundle_version_label:
-            versions = [float(version) for version in bundle_version.split(",")]
+            versions = [correct_float_precision(version) for version in bundle_version.split(",")]
             if float_valid_ocp_version >= versions[0]:
                 return True
-        elif float_valid_ocp_version >= float(bundle_version):
+        elif float_valid_ocp_version >= correct_float_precision(bundle_version):
```

Comment:
I don't think this will work as expected:

```python
>>> '4.10' > '4.9'
False
```

Reply:
I think you'll need something like this: https://gist.github.com/limed/473a498641bbc7761a20

Reply:
I see. We can't compare those numbers